### PR TITLE
ENH: Increase `itk::STAPLEImageFilter` coverage

### PR DESCRIPTION
**Pull request by jhlegarreta**
- STYLE: Make flow control variables have a local scope 
- STYLE: Conform to ITK style in variable naming in tests
- STYLE: Declare variables closer to where they are instantiated
- ENH: Increase `itk::STAPLEImageFilter` coverage

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
-

### CODE REVIEW

File: Modules/Filtering/ImageCompare/test/itkSTAPLEImageFilterTest.cxx [hunk 1]
```diff
@@ -155,15 +155,13 @@ template <unsigned int VDimension>
...
-  size_t i;
-
   typename itk::ImageFileReader<InputImageType>::Pointer  reader;
   typename itk::ImageFileWriter<OutputImageType>::Pointer writer = itk::ImageFileWriter<OutputImageType>::New();
 
   size_t number_of_files = m_Files.size();
 
   // Set the inputs
-  for (i = 0; i < number_of_files; ++i)
+  for (size_t i = 0; i < number_of_files; ++i)
```
Comment (dzenanz) [inline Modules/Filtering/ImageCompare/test/itkSTAPLEImageFilterTest.cxx]: Commit message body is redundant with commit message summary (the first line).

**Review by thewtex — APPROVED** [review]
@jhlegarreta thanks!

File: Modules/Filtering/ImageCompare/test/itkSTAPLEImageFilterTest.cxx [hunk 2]
```diff
@@ -158,10 +158,10 @@ Stapler<VDimension>::Execute()
   typename itk::ImageFileReader<InputImageType>::Pointer  reader;
   typename itk::ImageFileWriter<OutputImageType>::Pointer writer = itk::ImageFileWriter<OutputImageType>::New();
 
-  size_t number_of_files = m_Files.size();
+  size_t numberOfFiles = m_Files.size();
```
Comment (dzenanz) [inline Modules/Filtering/ImageCompare/test/itkSTAPLEImageFilterTest.cxx:190]: Commit message body redundant with summary.

File: Modules/Filtering/ImageCompare/test/CMakeLists.txt [hunk 3]
```diff
@@ -28,10 +28,10 @@ itk_add_test(NAME itkConstrainedValueDifferenceImageFilterTest
 itk_add_test(NAME itkSimilarityIndexImageFilterTest
       COMMAND ITKImageCompareTestDriver itkSimilarityIndexImageFilterTest)
 itk_add_test(NAME itkSTAPLEImageFilterTest
-      COMMAND ITKImageCompareTestDriver
-    --compare DATA{${ITK_DATA_ROOT}/Baseline/Algorithms/STAPLEImageFilterTest.mha}
-              ${ITK_TEST_OUTPUT_DIR}/STAPLEImageFilterTest.mha
-    itkSTAPLEImageFilterTest 2 ${ITK_TEST_OUTPUT_DIR}/STAPLEImageFilterTest.mha 255 0.5 DATA{${ITK_DATA_ROOT}/Input/STAPLE1.png} DATA{${ITK_DATA_ROOT}/Input/STAPLE2.png} DATA{${ITK_DATA_ROOT}/Input/STAPLE3.png} DATA{${ITK_DATA_ROOT}/Input/STAPLE4.png})
+  COMMAND ITKImageCompareTestDriver
```
Comment (dzenanz) [inline Modules/Filtering/ImageCompare/test/CMakeLists.txt:31]: 👍

**Review by dzenanz — APPROVED** [review]
Minor remarks inline.

File: Modules/Filtering/ImageCompare/test/itkSTAPLEImageFilterTest.cxx (hunk 2 above)
Comment (jhlegarreta) [reply Modules/Filtering/ImageCompare/test/itkSTAPLEImageFilterTest.cxx:190]: I always put a commit body; event for commits where the subject line is sufficiently descriptive.

**Comment by jhlegarreta** [comment]
@dzenanz Concerning the commit body messages, I have always followed the guidelines
https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#commit-messages

As said, I always add a body, even when the changes are small enough to be described by the subject line. And at times, the body just restates the subject line.

**Review by hjmjohnson — APPROVED** [review]

**Comment by jhlegarreta** [comment]
Looks like this introduced a Valgrind defect:
https://open.cdash.org/viewDynamicAnalysisFile.php?id=9746227

My hope is that #3626 will fix this.

**Pull request merged**